modules/igv/1.0/igv.smk

```diff
@@ -0,0 +1,824 @@
+
+def get_maf(wildcards):
+    unix_group = config["unix_group"]
+    return expand(config["lcr-modules"]["igv"]["inputs"]["maf"], allow_missing=True, unix_group=unix_group)
+
+
+##### RULES #####
+
+
+
+# Symlinks the input files into the module results directory (under '00-inputs/')
+rule _igv_symlink_bams:
+    input:
+        bam = get_bam,
+        bai = get_bai
+    output:
+        bam = CFG["dirs"]["inputs"] + "bams/{seq_type}/{sample_id}.bam",
+        bai = CFG["dirs"]["inputs"] + "bams/{seq_type}/{sample_id}.bam.bai"
+    run:
+        op.absolute_symlink(input.bam, output.bam)
+        op.absolute_symlink(input.bai, output.bai)
+
+rule _igv_symlink_maf:
+    input:
+        maf = get_maf
+    output:
+        maf = CFG["dirs"]["inputs"] + "maf/{seq_type}--{genome_build}/{tumour_id}--{normal_sample_id}--{pair_status}.maf"
+    run:
+        op.absolute_symlink(input.maf, output.maf)
+
+# Reduce MAF columns to prevent parsing errors in Pandas and reduce file size
+rule _igv_reduce_maf_cols:
+    input:
+        maf = str(rules._igv_symlink_maf.output.maf)
+    output:
+        maf = temp(CFG["dirs"]["inputs"] + "maf/{seq_type}--{genome_build}/{tumour_id}--{normal_sample_id}--{pair_status}.maf.temp")
+    shell:
+        op.as_one_line("""
+        cut -f 1,5,6,7,9,10,11,13,16,17 {input.maf} > {output.maf}
+        """)
+
+rule _igv_merge_regions:
+    input:
+        input_regions = lambda w: config["lcr-modules"]["igv"]["regions"][w.tool_type][w.tool_build]
+    output:
+        merged_regions = CFG["dirs"]["inputs"] + "regions/{tool_type}_merged.{tool_build}.tsv"
+    log:
+        stdout = CFG["logs"]["inputs"] + "regions/merge_{tool_type}_regions.{tool_build}.stdout.log",
+        stderr = CFG["logs"]["inputs"] + "regions/merge_{tool_type}_regions.{tool_build}.stderr.log"
+    run:
+        merged_df = pd.DataFrame()
+        for result in input.input_regions:
+            try:
+                df = pd.read_table(result, comment="#", sep="\t")
+                merged_df = pd.concat([merged_df, df])
+            except:
+                with open(log.stdout, "a") as header:
+                    header.write(f"Error reading or merging file {result}\n")
+        merged_df.to_csv(output.merged_regions, sep="\t", na_rep="NA", index=False)
+
+# Convert input regions file into BED format
+rule _igv_format_regions:
+    input:
+        regions = str(rules._igv_merge_regions.output.merged_regions)
+    output:
+        regions = CFG["dirs"]["inputs"] + "regions/{tool_type}_formatted.{tool_build}.tsv"
+    params:
+        regions_format = lambda w: w.tool_type,
+        regions_build = lambda w: w.tool_build
+    log:
+        stdout = CFG["logs"]["inputs"] + "regions/format_regions_{tool_type}.{tool_build}.stdout.log",
+        stderr = CFG["logs"]["inputs"] + "regions/format_regions_{tool_type}.{tool_build}.stderr.log"
+    script:
+        config["lcr-modules"]["igv"]["scripts"]["format_regions"]
+
+REGIONS_FORMAT = {
```

Comment:
Why do we need a dictionary if everything is mapped to the same key `bed`?  Maybe we can just refer to bed regardless of what is the region format?

Reply:
You're right- I removed it